Feature/nationality

iSamity opened the pull request.

zekew11 committed:
add nationality submod

zekew11 committed:
working on nationality

Humoresque811 committed:
Merging in changes for GA from stable

Merging in all changes on stable since branch created, including GA, stability, versioning changes, to prep for work on nationality with up to date features.

Humoresque811 committed:
Popped previous stashed changes with work on nationality

Changes included some progress on nationality feature

Humoresque811 committed:
Minor naming, structure updates

Updated names to be significantly more logical, moved things to folders

Humoresque811 committed:
New flight generator base system

Completely replaced old format with a new generator format, where the result for a flight generator is much clearer, and some logic is handled in a abstract base flight generator. Rigged up the default flight generator class to work with the update

Humoresque811 committed:
Started Nationality Route Generator

Added some basic content, prep for full buildout

Humoresque811 committed:
Added Nationality Features, Unit Testing

Added a rough but theoretically functional (im sure its not though) version of nationality flight generation. Added an xUnit unit testing project as well, currently not well used

iSamity commented on AirportCEONationality/AirportCEONationality.cs:
```diff
@@ -0,0 +1,62 @@
+    [BepInDependency("org.airportceotweakscore.zeke")]
+    public class AirportCEONationality : BaseUnityPlugin
+    {
+        public const string GUID = "org.airportceonationality.zeke";
```
this is all you man

iSamity commented on AirportCEONationality/AirportCEONationality.csproj:
```diff
@@ -0,0 +1,68 @@
+    <AssemblyName>AirportCEONationalityk</AssemblyName>
+	<Product>AirportCEONationality</Product>
+    <Description></Description>
+	<Version>3.0.809.131</Version>
```
weird version

iSamity commented on AirportCEONationality/ModLoaderInteractionHandler.cs:
```diff
@@ -0,0 +1,27 @@
+            // More will probably be added!
+            AirportCEONationality.LogInfo("Seting up ModLoader interactions");
+
+            WatermarkUtils.Register(new WatermarkInfo("T-N", Assembly.GetExecutingAssembly().GetName().Version.ToString(), true));
```
I would use the new way of uusing verison instead of using assembly see other projects examples

iSamity commented on AirportCEONationality/RouteGen/NationalityFlightGenerator.cs:
```diff
@@ -0,0 +1,347 @@
+
+        // Preselect route number ...............................................................................................
+
+        int maxFlightNumber = (((int)airlineModel.businessClass + 3) ^ 2) * 50 + Utils.RandomRangeI(100f, 200f);
```
explain this formula

Humoresque811 replied:
I have no idea its what the old tweaks system and maybe the game uses...

iSamity commented on AirportCEONationality/RouteGen/NationalityFlightGenerator.cs:
```diff
@@ -0,0 +1,347 @@
+
+            Route inboundRouteF = new Route(route.route);
+            Route outboundRouteF = new Route(inboundRouteF);
+            outboundRouteF.ReverseRoute();
```
why do you need to reverse the route here?

Humoresque811 replied:
The inbound and outbound flights of an aircraft (which are always the same ig, at least thats how it was in old tweaks) take separate routes

iSamity commented on AirportCEONationality/RouteGen/NationalityFlightGenerator.cs:
```diff
@@ -0,0 +1,347 @@
+
+    public bool FleetMemberCanServeRoute(AirlineFleetMember fleetMember, RouteContainer route, float chanceToOfferRegaurdless = 0, bool debug = false)
+    {
+        if (debug)
```
maybe use this instead of extra properties you have to pass to debug

https://stackoverflow.com/questions/2104099/c-sharp-if-then-directives-for-debug-vs-release

if you build in debug it will log and if you build it in release it wont

iSamity commented on AirportCEOTweaksCore/Airline/IServeAircraftTypeCheck.cs:
why name it like this, with the I

iSamity replied:
nvm i see because you put an interface in the file

Humoresque811 replied:
This is from Zekews code btw, not mine lol

Humoresque811 committed:
Updates to Nationality, tried to get things loading

Updated versioning format for nationality, also tried to get airline model working but failed

Humoresque811 committed:
Working nationality flight generator

!!!!!!!! Should work without issues, however it was just tested with an international airline so far - potential changes still pending

Humoresque811 committed:
Merged main

Humoresque811 committed:
Updated AircraftAdder

with post-merge changes

Humoresque811 committed:
Established proper project scope

Updated sln to included appropriate projects but nothing more.

Humoresque811 committed:
New fleet creation system, fixes

Fixed finally issues in Aircraft, fixed all nationality bugs in core.

Humoresque811 committed:
Replaced Suitability System

Updated suitability system to a new one that should be easier to mess with (based on graphs and clearer logic). Overall fixes

Humoresque811 committed:
Adding warnings and customizability

Adding warnings if flight generators are unable to generate a flight (optionable, customizable), and added options to control fallback of nationality

Humoresque811 committed:
Nationality: Added logs, reduced international flights

Weighted down international flights in nationality and added debug logs to actual show whats going on - two new configs as well.

iSamity commented:
used to review the files
